uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -0,0 +1,64 @@
+define([
+  'leave-absences/shared/modules/apis',
+  'common/services/api'
+], function (apis) {
+  'use strict';
+
+  apis.factory('LeaveRequestAPI', ['$log', 'api', function ($log, api) {
+    $log.debug('LeaveRequestAPI');
+
+    return api.extend({
+
+      /**
+       * This method returns all the Leave Requests.
+       * It supports filters, pagination, sort and extra params
+       *
+       * @param {object} filters - Values the full list should be filtered by
+       * @param {object} pagination
+       *   `page` for the current page, `size` for number of items per page
+       * @param {string} sort - The field and direction to order by
+       * @param  {Object} params
+       * @return {Promise}
+       */
+      all: function (filters, pagination, sort, params) {
+        $log.debug('LeaveRequestAPI.all');
+
+        return this.getAll('LeaveRequest', filters, pagination, sort, params, 'getFull')
+          .then(function (data) {
+            return data;
+          });
+      },
+
+      /**
+       * This method returns all the total change in balance that is caused by the
+       * leave requests of a given absence type, or of all the absence types of a given contact and period.
+       *
+       * @param {string} contactId Mandatory The ID of the Contact to get the balance change for
```

Comment:
If a param is mandatory, there is no need to specify it. Also I think in this case both for `contactid` and `periodId` the name of the param is self-explanatory, no need to leave a comment
```js
@param {string} contactId
@param {string} periodId
```